server/src/test/java/io/crate/integrationtests/DynamicMappingUpdateITest.java

```diff
@@ -64,6 +66,22 @@ insert into t values (1, {x=1})
                 """);
         execute("refresh table t");
 
+        AtomicBoolean dmlStatementsFinished = new AtomicBoolean(false);
+        HashMap<String, Integer> columnPositions = new HashMap<>();
+        Thread checkPositions = new Thread(() -> {
+            while(dmlStatementsFinished.get() == false) {
+                synchronized (response) {
+                    execute("select column_name, ordinal_position from information_schema.columns where table_name = 't'");
+                    String columnName = (String) response.rows()[0][0];
+                    columnPositions.computeIfPresent(columnName, (key, existingValue) -> {
+                        throw new IllegalStateException(
+                            String.format(Locale.ENGLISH, "Column %s had position %d which is recomputed to %d", columnName, existingValue, response.rows()[0][1])
```

Comment:
Thank you @BaurzhanSakhariev, this looks good for me. 
One minor thing, don't you need
```
if (existingValue != response.rows()[0][1]) 
```
then throw?
Also, `put` calls seems missing?

Reply:
Oh and probably `columnPositions = new ConcurrentHashMap<>();`

Reply:
Good catch, thanks!

I somehow messed up `computeIfPresent` functionality and thought it will do everything you mentioned. Rebased with put and !equals call